src/Microsoft.Azure.WebJobs.Host/Config/ConcurrencyOptionsSetup.cs

```diff
@@ -0,0 +1,89 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.Azure.WebJobs.Host.Scale;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.Azure.WebJobs.Host.Config
+{
+    internal class ConcurrencyOptionsSetup : IConfigureOptions<ConcurrencyOptions>
+    {
+        private const int BytesPerGB = 1024 * 1024 * 1024;
+
+        public void Configure(ConcurrencyOptions options)
+        {
+            string sku = Utility.GetWebsiteSku();
+            int numCores = Utility.GetEffectiveCoresCount();
+            Configure(options, sku, numCores);
+        }
+
+        internal static void Configure(ConcurrencyOptions options, string sku, int numCores)
+        {
+            long memoryLimitBytes = GetMemoryLimitBytes(sku, numCores);
+            if (memoryLimitBytes > 0)
+            {
+                // if we're able to determine the memory limit, apply it
+                options.TotalAvailableMemoryBytes = memoryLimitBytes;
+            }
+        }
+
+        internal static long GetMemoryLimitBytes(string sku, int numCores)
+        {
+            if (!string.IsNullOrEmpty(sku))
+            {
+                float memoryGBPerCore = GetMemoryGBPerCore(sku);
+
+                if (memoryGBPerCore > 0)
+                {
+                    double memoryLimitBytes = memoryGBPerCore * numCores * BytesPerGB;
+
+                    if (string.Equals(sku, "IsolatedV2", StringComparison.OrdinalIgnoreCase) && numCores == 8)
+                    {
+                        // special case for upper tier IsolatedV2 where GB per Core
+                        // isn't cleanly linear
+                        memoryLimitBytes = (float)23 * BytesPerGB;
```

Comment:
For some reason, Isolatedv2 doesn't continue the 4GB/core pattern for 8 core. See https://azure.microsoft.com/en-us/pricing/details/app-service/windows/